cohortextractor/tpp_backend.py

```diff
@@ -2720,6 +2721,41 @@ def patients_outpatient_appointment_date(
               OPA.Patient_ID
             """
 
+    def patients_with_value_from_file(self, f_path, returning, type_):
+        table_name = self.get_temp_table_name("file")
+        column_name = self._current_column_name
+
+        values = pandas.read_csv(
+            f_path,
+            index_col="patient_id",
+            usecols=["patient_id", column_name],
+            squeeze=True,
+        )
+
+        queries = [
+            f"""
+            -- Uploading file for {column_name}
+            CREATE TABLE {table_name} (
+                patient_id VARCHAR(MAX),
+                {column_name} VARCHAR(MAX)
+            )
+            """,
+        ]
+        insert_sql = f"INSERT INTO {table_name} (patient_id, {column_name}) VALUES"
+
+        batch_size = 999
+        for i in range(0, len(values), batch_size):
+            values_batch = values.iloc[i : i + batch_size]
+            values_sql_lines = [
+                "({}, {})".format(*map(quote, row)) for row in values_batch.items()
+            ]
+            values_sql = ",\n".join(values_sql_lines)
+            queries.append(f"{insert_sql}\n{values_sql}")
```

Comment:
I think you're missing the final `SELECT` query here, which is why it's complaining. You can look at the other `patients_` methods for inspiration, but the basic pattern is that the final query in the list of returned queries must be the one which selects the column data. From a quick look at the code I think that might be something like this:
```sql
SELECT
  patient_id,
 {column_name} AS {returning}
FROM
  {table_name}
```

Reply:
Thanks @evansd. I added the final `SELECT` and MSSQL stopped complaining.